### PR TITLE
Add Honeybadger team terraform resource

### PR DESCRIPTION
**Pull request by miquelbar**
Add/Remove/Update/Invite into Honeybadger teams or users using terraform resource

### CODE REVIEW

File: cli/team.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	err = json.Unmarshal(body, &hbTeams)
+	if err != nil {
+		return hbTeams.Teams, err
+	}
+
+	return hbTeams.Teams, nil
+}
+
+// FindTeamByName - Find Team by name
+func (hbc *HoneybadgerClient) FindTeamByName(teamName string) (HoneybadgerTeam, error) {
```
Comment (mpucholblasco) [inline cli/team.go:35]: If you return an object, the object is copied from the `return` command. If the object to be returned is new (created inside the method, as it's in this case), I guess it's better to return just the pointer.

```suggestion
func (hbc *HoneybadgerClient) FindTeamByName(teamName string) (*HoneybadgerTeam, error) {
```

In this way, you can then return just a `nil` on error cases.

File: honeybadger/data_source_teams.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+		unstructuredTeam["name"] = team.Name
+		unstructuredTeam["id"] = team.ID
+		unstructuredTeam["created_at"] = team.CreatedAt
+		for _, user := range team.Users {
+			unstructuredUser := make(map[string]interface{})
+			unstructuredUser["id"] = user.ID
+			unstructuredUser["name"] = user.Name
+			unstructuredUser["email"] = user.Email
+			unstructuredUser["admin"] = user.IsAdmin
```
Comment (mpucholblasco) [inline honeybadger/data_source_teams.go]:
```suggestion

			unstructuredUser := map[string]interface{}{
				"id": user.ID,
				"name": user.Name,
				"email": user.Email,
				"admin": user.IsAdmin,
			}
```

File: honeybadger/data_source_teams.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+	var unstructuredTeams []map[string]interface{}
+
+	for _, team := range teams {
+		var unstructuredUsers []map[string]interface{}
+		unstructuredTeam := make(map[string]interface{})
+
+		unstructuredTeam["name"] = team.Name
+		unstructuredTeam["id"] = team.ID
+		unstructuredTeam["created_at"] = team.CreatedAt
```
Comment (mpucholblasco) [inline honeybadger/data_source_teams.go]:
```suggestion
		unstructuredTeam := map[string]interface{}{
			"name": team.Name,
			"id": team.ID,
			"created_at": team.CreatedAt,
		}
```

File: honeybadger/provider.go [hunk 4]
```diff
@@ -22,17 +22,13 @@ func Provider() *schema.Provider {
...
 			},
-			"team_id": &schema.Schema{
-				Type:        schema.TypeInt,
-				Optional:    true,
-				DefaultFunc: schema.EnvDefaultFunc("HONEYBADGER_TEAM_ID", nil),
-			},
 		},
 		DataSourcesMap: map[string]*schema.Resource{
-			"honeybadger_users": dataSourceUsers(),
+			"honeybadger_teams": dataSourceTeams(),
```
Comment (mpucholblasco) [inline honeybadger/provider.go:27]: If you perform this change, aren't we loosing the datasource related to users?

**Review by mpucholblasco — APPROVED** [review]

File: honeybadger/provider.go (hunk 4 above)
Comment (miquelbar) [reply honeybadger/provider.go:27]: No, `honeybadger_teams`gets also  the user information. Every teams have more than one member

File: cli/team.go (hunk 1 above)
Comment (miquelbar) [reply cli/team.go:35]: Good to know. I'll keep in mind 😄

**Review by treezio — APPROVED** [review]
👏🏻 👏🏻 👏🏻

**Review by mpucholblasco — APPROVED** [review]

**Pull request merged**